### ISSUE TITLE
Stack overflow error when scaling

### ISSUE THREAD

**Issue by crbinz** [JuliaStats/PDMats.jl#50]
```
julia> a = PDMat(eye(3))
PDMats.PDMat{Float64,Array{Float64,2}}(3,[1.0 0.0 0.0; 0.0 1.0 0.0; 0.0 0.0 1.0],Base.LinAlg.Cholesky{Float64,Array{Float64,2}} with factor:
[1.0 0.0 0.0; 0.0 1.0 0.0; 0.0 0.0 1.0])

julia> 3 * a
ERROR: StackOverflowError:
 in *(::PDMats.PDMat{Float64,Array{Float64,2}}, ::Int64) at /Users/cbinz/.julia/v0.5/PDMats/src/generics.jl:24 (repeats 80000 times)
```

This works as expected when the scaling factor is a `Float64`, by the way.

**Comment by crbinz** [comment]
[Here](https://github.com/JuliaStats/PDMats.jl/blob/master/src/pdmat.jl#L37), this will happen any time the argument of `c` is not the same type as `a`.  I think this could just be changed like this: 

```
*{S<:Real, T<:Real}(a::PDMat{S}, c::T) = PDMat(a.mat * c)
```

along with the relevant changes [here](https://github.com/JuliaStats/PDMats.jl/blob/master/src/generics.jl#L24). However, I don't know enough to say whether this is a good idea or not, or what other implications this might have.

**Comment by crbinz** [comment]
Closed by #54
